packages/web-components/src/presence-badge/presence-badge.ts

```diff
@@ -0,0 +1,78 @@
+import { FASTElement, attr } from '@microsoft/fast-element';
+import {
+  statusAvailableGlyph,
+  statusAwayGlyph,
+  statusBusyGlyph,
+  statusDndGlyph,
+  statusOfflineGlyph,
+  statusOofGlyph,
+  statusUnknownGlyph,
+  statusAvailableOofGlyph,
+  statusBusyOofGlyph,
+  statusDndOofGlyph,
+  statusBlockedGlyph,
+} from './svg';
+
+type SlotInfo = { name: string; icon: string };
+
+export class PresenceBadge extends FASTElement {
+  /**
+   * Displays the user's current status as an svg icon.
+   * @public
+   * @remarks
+   * HTML Attribute: status
+   */
+  @attr
+  public status: string;
+
+  /**
+   * The user's current out-of-office status, which modifies or overrides their status.
+   * @public
+   * @remarks
+   * HTML Attribute: outofoffice
+   */
+  @attr({ mode: 'boolean' })
+  public outOfOffice: boolean;
+
+  public getSlotInfo(): SlotInfo {
+    if (!this.outOfOffice) {
```

Comment:
I think you can reduce the emitted size of this function pretty significantly by refactoring the `this.outOfOffice` check to be inside the switch:

```ts
public getSlotInfo(): SlotInfo {
	const oofSwitch = (normal, oof) => this.outOfOffice ? oof : normal;
      switch (this.status) {
        case 'available':
          return { name: this.status, icon: offSwitch(statusAvailableGlyph, statusAvailableOofGlyph) };
        case 'away':
          return { name: this.status, icon: offSwitch(statusAwayGlyph, statusOofGlyph) };
        case 'busy':
          return { name: this.status, icon: offSwitch(statusBusyGlyph, statusBusyOofGlyph) };
        case 'dnd':
          return { name: this.status, icon: offSwitch(statusDndGlyph, statusDndOofGlyph) };
        case 'offline':
          return { name: this.status, icon: offSwitch(statusOfflineGlyph, statusOofGlyph) };
        case 'outofoffice':
          return { name: this.status, icon: statusOofGlyph };
        case 'blocked':
          return { name: this.status, icon: statusBlockedGlyph };
        default:
          return { name: 'unknown', icon: statusUnknownGlyph };
      }
  }
```

Reply:
Actually that would make the binding volatile, which could lead it not to update. You would need to refactor to have "name" and "icon" be independent observable values. Feel free to disregard this